src/Couchbase.Lite.Tests.Shared/DatabaseTest.cs

```diff
@@ -305,6 +307,104 @@ public void TestSaveConcurrencyControl()
             }
         }
 
+        [Fact]
+        public void TestConflictHandlerSaveMergedDocument()
+        {
+            using (var doc1 = new MutableDocument("doc1")){
+                doc1.SetString("name", "Jim");
+                
+                Db.Save(doc1);
+
+                // Get two doc1 document objects (doc1a and doc1b):
+                var doc1a = Db.GetDocument(doc1.Id).ToMutable();
+                var doc1b = Db.GetDocument(doc1.Id).ToMutable();
+
+                // Modify doc1a:
+                doc1a.SetString("name", "Jim");
+                doc1a.SetString("language", "English");
+                        
+                Db.Save(doc1a);
+
+                // Modify doc1b:
+                doc1b.SetString("name", "Jim");
+                doc1b.SetString("language", "C#");
+                doc1b.SetString("location", "Japan");        
+                Db.Save(doc1b, ResolveConflict);
+            }
+
+            using (var doc1 = Db.GetDocument("doc1")) {
+                doc1.GetString("name").Should().Be("Jim");
+                var lanStr = doc1.GetString("language");
+                lanStr.Should().Contain("English");
+                lanStr.Should().Contain("C#");
+                doc1.GetString("location").Should().Be("Japan");
+            }
+        }
+
+        [Fact]
+        public void TestConflictHandlerReturnsTrue()
+        {
+            using (var doc1 = new MutableDocument("doc1")) {
+                doc1.SetString("name", "Jim");
+                Db.Save(doc1, (updated, current) => {
+                    return true;
+                });
+
+                Db.GetDocument("doc1").GetString("name").Should().Be("Jim");
+
+                var doc1a = new MutableDocument(doc1.Id);
+                doc1a.SetString("name", "Kim");
+                Db.Save(doc1a, (updated, current) => {
+                    return true;
+                });
+
+                Db.GetDocument("doc1").GetString("name").Should().Be("Kim");
+            }
+        }
+
+        [Fact]
+        public void TestConflictHandlerReturnsFalse()
+        {
+            using (var doc1 = new MutableDocument("doc1")) {
+                doc1.SetString("name", "Jim");
+                Db.Save(doc1, (updated, current) => {
+                    return false;
+                });
+
+                Db.GetDocument(doc1.Id).GetString("name").Should().Be("Jim");
+
+                var doc1a = new MutableDocument(doc1.Id);
+                doc1a.SetString("name", "Kim");
+                Db.Save(doc1a, (updated, current) => {
+                    return false;
+                });
+
+                Db.GetDocument("doc1").GetString("name").Should().Be("Jim");
+            }
+        }
+
+        [Fact]
+        public void TestConflictHandlerWithDeletedOldDoc()
+        {
+            using (var doc1 = new MutableDocument("doc1")) {
```

Comment:
This does not test "deleted old doc".  In order to do that you'd need to delete and then try to save with the conflict handler.